### ISSUE TITLE
3.6.1 not updating coordinates continuously

### ISSUE THREAD

**Issue by smartroad** [zapmaker/GrblHoming#78]
Maybe this isn't a bug, but when sending a file the machine and work coordinates only seem to update after a long move and not continuously as the cutter moves (compiled for use on my Raspberry Pi). I only mention as when I use a UniversalGcodeSender (on my PC I admit) it updates the coordinates constantly. I changed the Display setting to "Always Request but without Idle Check" but no change.

Is it because I am using a RPi?

**Comment by gerritv** [comment]
Grbl echos the position after each completed gcode. I think a useful modification to this wonderful tool would be to use that instead of polling for the position. Or guessing where the tool might be. I suspect this would simplify the visualization code as well? I'll be looking at the source code to ee if I can help in some way. The overall GUI of this tool is better than some of the others. IMO

**Comment by LETARTARE** [comment]
@gerritv 
I did a test to predict the position of the tool in "https://github.com/zapmaker/GrblHoming/tree/Letartare" :
You can try 'Tools->Options->Display->Synchronous simulation'.
This is experimental !!

**Comment by gerritv** [comment]
Thank you, I have done some testing. I really like the 3D visualization but there are some behavioural issues overall with the tool. I would help but I only use Visual Studio for development, I guess I will have to figure out how to compile using that.
 I will write up my observations later today.

**Comment by LETARTARE** [comment]
@gerritv 
thank you for trying this code.
=====>  I just noticed that I forgot to work with G20 !! Only use files using G21. <=====
Under Win32, usually after reading 'buildWin.txt', you can use 'mw.bat'.

**Comment by gerritv** [comment]
@LETARTARE yes, figured out the G21 :-) For dev I need Visual Studio, I no longer hack at things with a raw environment, life is too short. This way I can help if I decide this is the gui for me and grbl.

**Comment by gerritv** [comment]
@LETARTARE Spent several hours on QT install, builds etc. Too many issues just getting a dev environment that I can work with, sorry.
I will give my comments on the app behaviour later. Need a drink now.

**Comment by LETARTARE** [comment]
I just introduced a new version
https://github.com/zapmaker/GrblHoming/tree/Letartare

**Comment by ghost** [comment]
It looks like grbl (v1.1f.20170801) is not sending back data that can be matched by the regex.  See gcode.cpp parseCoordinates() method.
A typical response to '?' is <Idle|MPos:0.000,0.000,2.000|FS:0,0> the OLD  grbl wiki states the return should look like <Idle,MPos:0.000,0.000,0.000,WPos:0.000,0.000,0.000>
A regex that tries to match the OLD documented string will never match the actual return string so the GUI coordinates will never be updated.

The OLD wiki: https://github.com/grbl/grbl/wiki//Interfacing-with-Grbl#grbl-response-meanings
The NEW wiki: https://github.com/gnea/grbl/wiki/Grbl-v1.1-Interface
So it looks like the regex matching in parseCoordinates() needs updating to suite later grbl code.

Additionaly to get both MPos and WPos grbl setting $10 needs to set to either 1 OR 2 to get either the MPos or WPos data. It no longer sends both in a single return string. So two calls will need to made to set $10 and then get the required data.
